addons/payment_razorpay_oauth/models/payment_provider.py

```diff
@@ -131,6 +131,19 @@ def action_razorpay_create_webhook(self):
             },
         }
 
+    # === CONSTRAINT METHODS === #
+
+    @api.constrains('state')
+    def _check_razorpay_credentials_are_set_before_enabling(self):
+        """ Check that the Razorpay credentials are valid.
+
+        :raise ValidationError: If the Razorpay credentials are not valid.
+        """
+        for provider in self.filtered(lambda p: p.code == 'razorpay' and p.state == 'enabled'):
+            if not provider.razorpay_key_id or not provider.razorpay_key_secret:
+                raise ValidationError(_("Razorpay credentials are missing. Click the \"Connect\""
+                                        "button to set up your account"))
```

Comment:
```suggestion
                raise ValidationError(_(
                    "Razorpay credentials are missing. Click the \"Connect\" button to set up your"
                    " account"
                ))
```
also to fix the missing space after Connect.